### ISSUE TITLE
Introduce The public API of Munzee in To Live Map

### ISSUE THREAD

**Issue by Rynee** [cgeo/cgeo#2093]
after speaking with mucek4 after receiving an update to the Cachesense app where Munzee was added it was suggested to me to open this issue. Munzee will give access to the Public API as well as CSV files needed to display Munzee geolocations. Cgeo is by far the one of the top Geocaching apps and with introducing Munzee API is can be the one app to rule them all 

_Sent via Hubroid_

**Comment by Lineflyer** [comment]
Reopening this (too long) thread as we have new information and the last statement was, that we can reopen if some conditions are met.

Still we need someone willing and able to implement it if the below mentioned is technically compatible at all.

---
Howdy xxx,

My name is Scott xxx and I am the Co-Founder and VP of Technology for Munzee. We had talked briefly last year about integrating Munzee within c:geo and unfortunately, at the time, we weren't able to provide good documentation or implementation. We have spent the past few weeks working to solve that problem so that Munzee information can be integrated within c:geo.

We have an easy to use OAuth 2.0 API that can be used to get munzees within a bounding box (lat/lng), munzee information, player information, leaderboard information, etc.

You can start reading more information at the following links

http://www.munzee.com/api
http://www.munzee.com/api/bearer/
http://www.munzee.com/api-docs/

Please let me know what I can do for this to happen.

Thank you,
Scott xxx

---

**Comment by SammysHP** [comment]
And I'll still request integration of OSM notes when we integrate Munzee in the c:geo core (as usually I'm mapping while geocaching).

Oh, and http://confluence.org/ of course.

@Bananeweizen wrote:
> After all, that is not much different from virtual caches. And we already support waymarks.

Where can I access Waymarks data in c:geo? It might be possible to import a GPX file, but that's already possible with Munzees.

**Comment by Lineflyer** [comment]
> And I'll still request integration of OSM notes when we integrate Munzee in the c:geo core (as usually I'm mapping while geocaching).

> Oh, and http://confluence.org/ of course.

Might be you are a little sarcastic? But hinestly people are different.

I think the jidgement of whether we want something like that in c:geo should be focussed on "Its a game" and "It is using GPS to find a target and log it (on site and) online"

So, that would be: Munzee: OK, OSM notes: NOK, confluence.org = OK (if they have an API)

**Comment by SammysHP** [comment]
Then "c:geo - Geocaching for Android" won't be correct anymore. Maybe "c:geo - Geogaming for Android". ;)

**Comment by Lineflyer** [comment]
> Then "c:geo - Geocaching for Android" won't be correct anymore. Maybe "c:geo - Geogaming for Android". ;)

Maybe.... and does not sound so bad.
Anyway for me Munzee integration is nothing really important from my perspective...but If someone comes around offering to implement it, I would not say No (and at least @Bananeweizen seems to also agree, not sure about others from the team).

**Comment by SammysHP** [comment]
Well, I can live with with Munzee integration, but we need someone who will maintain it (for more than a month or two).

**Comment by Bananeweizen** [comment]
I believe it is actually really easy to just query for munzees and to display them (in lists and maps). What will be way more complex is actually making the user able to do munzee specific stuff, e.g. scanning the QR code, logging, displaying different types of munzees and so on.

Therefore I would only start with integrating the query and display part, just like we do it for all kinds of connectors right now.

**Comment by Lineflyer** [comment]
> Therefore I would only start with integrating the query and display part, just like we do it for all kinds of connectors right now.

I totally agree. The whole logging part with camera support and QRcode reading is way too special for the first step and need good thoughts about whether it makes sene at all to integrate this into c:geo itself (or just refer to another app or a plugin).

**Comment by Rynee** [comment]
As far as capturing that would still probably need to be handled by the
actually Munzee app a capture icon could be included to display and when
pressed a call to open the Munzee app would be made

**Comment by rsudev** [comment]
I would prefer to have such a support rather as a plugin, as I see this at least as sensitive as Contacts or Calendar interaction. Munzee explicitly states that they capture, record and analyze location information and may disclose that even to third parties for example for their advertisement business.
Something like that should be very explicit opt-in and therefore be best supplied through a separate plugin.

**Comment by akkatracker** [comment]
@rsudev I agree. Munzee is a very commercial game (I don't 100% agree with their leadership or practices). That said if cgeo was to implement it (as many people want) then plugin would be a better option imo - it also helps seperate cgeo for Geocaching and "cgeo for geogames"

**Comment by Rynee** [comment]
I have to ask what's wrong with the leadership and practices ?

**Comment by akkatracker** [comment]
They made a huge cash grab and stuffed up a large portion of the game's dynamics. You've now got a huge issue where Mystery virtuals yield more points than most other stuff and they're virtual and able to be deployed anywhere. The amount of points you can get is only limited by the size of your wallet- Munzee was initially intended as a physical game. Groundspeak made the right choice grandfathering virtuals - a game's gotta make up its mind. Continuing on the way it's going I'd almost place money that in 3 years Munzee won't be around - there's gotta be huge changes. 

Cgeo is open source so if data is used by someone else for commercial purposes that's a bit iffy. Needing to open an external app to actually scan makes the function a better fit for a plugin. 

Each to their own though but @rsudev 's proposal is the best IMO - it gives Munzee players an option for implementing it with cgeo (which when I played actively was a feature I would've found fun) as well as keeping cgeo in its roots a geocaching app.

**Comment by Rynee** [comment]
Have to say I'm amused at the fact that in your opinion they won't be
around in 3 years funny thing is people said that about them not even
lasting a year and here they are celebrating their 5th birthday

**Comment by akkatracker** [comment]
Be my guest to believe what you want. In their first year they weren't cash
grabbing. Many of the communities on Facebook etc have gone sour.

This is an issue thread, not a whinge thread so let's return to the
implementation of Munzee Api

**Comment by Rynee** [comment]
What Facebook communities are you in

**Comment by coolant** [comment]
If you guys would like to have a phone/skype call to discuss how Munzee and cgeo can work together, I'm all for it. You can also directly email me at scott@munzee.com. I have certainly done a lot with the API to help the app MunzStat be a much better app and fill the gaps that the main Munzee app cannot due to our limited resources. I'd love to help out cgeo too.

**Comment by akkatracker** [comment]
@coolant is the API still paid? If so that would be incompatible with c:geo. Would be nice to have as a plugin - should help join the games together, not split them. 

Not too sure if anyone here's willing to build in the functionality at present but I'll tag @Lineflyer to make sure he knows and can pass it on.

**Comment by coolant** [comment]
The API is free up until a point - https://www.munzee.com/api/. We do this so our infrastructure doesn't become too overloaded. Although, we also can make exceptions to the rule 😉

As far as working together, I'm all for it. Let me know how I can help besides code it. I don't know Java. 😄

**Comment by Vike91** [comment]
I don't know if this would be a justified comment but I'd love to be able to play Munzee in the c:geo app.
